apps/api/src/services/agent-manager.service.ts

```diff
@@ -1368,6 +1369,31 @@ export class AgentManager {
     }
   }
 
+  async getAgentPnlForComp(agentId: string, competitionId: string) {
```

Comment:
I refactored this into a function so it can be used inside the `endCompetition` function, and the values can be saved instead of recalculating on each request.